### ISSUE TITLE
Compile on save support

### ISSUE THREAD

**Issue by angelozerr** [theia-ide/typescript-lsp#9]
tsserver provides support for compile on save https://github.com/Microsoft/TypeScript/blob/master/src/server/protocol.ts#L18 by providing `CompileOnSaveAffectedFileList` and `CompileOnSaveEmitFile` commands.

I'm consuming thoses commands in https://github.com/angelozerr/typescript.java and it works well. It should be fantastic if typescript-lsp could provide a custom LSP command for this feature.

**Comment by angelozerr** [comment]
@svenefftinge is there a feature that you will be aware to implement? I'm studying how to migrate https://github.com/angelozerr/typescript.java to lsp4e, but without this feature I cannot manage compile on save with Eclipse Builder (and manage progress monitor because with tsc we cannot do that)

**Comment by svenefftinge** [comment]
I'm currently too busy with other things, but would happily accept and review a PR :)

**Comment by angelozerr** [comment]
Great:) I must just convince @mickaelistria to migrate from javascript-typescript-langserver to typescript-language-server in Eclipse BlueSky https://github.com/mickaelistria/eclipse-bluesky/issues/22

A very good news with typescript-language-server is that 

 * I can consume https://github.com/angelozerr/tslint-language-service to have tslint marker and quick fix with lsp4e. I have not tested, but I think Angular Language Service will work too.
 * I can choose TypeScript version, it's a very good feature too.

**Comment by mickaelistria** [comment]
> I must just convince @mickaelistria to migrate from javascript-typescript-langserver to typescript-language-server in Eclipse BlueSky

The best way to convince me is to submit a patch and show me a demo of the benefits provided by this new LS compared to the current one. BlueSky would pragmatically take the one that feels the best for users.

**Comment by angelozerr** [comment]
> The best way to convince me is to submit a patch and show me a demo of the benefits provided by this new LS compared to the current one

Yes sure:) It is in my scope. There are some regression with typescript-language-server  like https://github.com/theia-ide/typescript-language-server/issues/15 so I would like to be sure that regression will fe fixed before doing a demo

**Comment by tsujp** [comment]
Was there any progress on this?
